src/rust-crypto/rust-crypto.ts

```diff
@@ -640,6 +642,43 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
         return new RustVerificationRequest(request, this.outgoingRequestProcessor, this.supportedVerificationMethods);
     }
 
+    /**
+     * Implementation of {@link CryptoApi#checkOwnCrossSigningTrust}
+     */
+    public async checkOwnCrossSigningTrust({ allowPrivateKeyRequests }: CheckOwnCrossSigningTrustOpts): Promise<void> {
+        // Download the device keys and add them into the olm machine
+        const queryResult = await this.downloadDeviceList(new Set([this.userId]));
+        await this.olmMachine.markRequestAsSent(
+            uuidv4(),
+            RustSdkCryptoJs.RequestType.KeysQuery,
+            JSON.stringify(queryResult),
+        );
+
+        if (allowPrivateKeyRequests) {
+            // Fetch cross signing keys from the secret storage and import them in the olm machine
+            const masterKey = await this.secretStorage.get("m.cross_signing.master");
+            const selfSigningKey = await this.secretStorage.get("m.cross_signing.self_signing");
+            const userSigningKey = await this.secretStorage.get("m.cross_signing.user_signing");
+
+            await this.olmMachine.importCrossSigningKeys(masterKey, selfSigningKey, userSigningKey);
+        }
+
+        // Get the current device
+        const device: RustSdkCryptoJs.Device = await this.olmMachine.getDevice(
+            this.olmMachine.userId,
+            this.olmMachine.deviceId,
+        );
+
+        // Verify the device and upload the cross signing signatures
+        const request: RustSdkCryptoJs.SignatureUploadRequest = await device.verify();
+        await this.http.authedRequest(
+            Method.Post,
+            "/_matrix/client/v3/keys/signatures/upload",
+            undefined,
+            request.body,
+        );
```

Comment:
surely this should be documented in the doc-comment in cryptoApi?

Reply:
In the doc-comment: `Verify the current device`. We should also tell that we are uploading the device signatures ?

Reply:
I would have thought so. "Verify" is quite different to "sign and upload".